tests/compatibility_tests/docker_manager.py

```diff
@@ -494,21 +494,53 @@ def prepare_volume_for_rollback(self, target_version: str, source_volume: str, t
             raise RuntimeError(
                 f"Failed to prepare volume {source_volume} for rollback using image {image_name}: {e}") from e
 
+    def _check_image_exists_on_dockerhub(self, image_name: str) -> bool:
+        """Check if a Docker image exists on DockerHub."""
+        try:
+            namespace, repo_tag = image_name.split("/")
+            repo, tag = repo_tag.split(":")
+        except ValueError:
+            raise ValueError(f"Invalid image name format: {image_name}. Expected format: namespace/repo:tag")
+
+        url = f"https://hub.docker.com/v2/repositories/{namespace}/{repo}/tags/{tag}"
+        response = requests.get(url)
+        return response.status_code == 200
+
+
     def pull_image_from_dockerhub(self, image_name: str):
         """
         Pull a Docker image using the Docker SDK.
 
+        Starting from 2.17.0, the image name will have a "-cloud" suffix. E.g., "marqoai/marqo:2.17.0-cloud".
+
         Args:
             image_name (str): The name of the Docker image to pull.
 
         Raises:
             RuntimeError: If the image cannot be pulled.
         """
+
+        variants = [
```

Comment:
nit: should this be in this PR or a different one? If this is intended, we can leave it in

Reply:
Yes, this was a PR that has been merged into the mainline. I cherry pick it into this to ensure the backwards compatibility tests can run